python/ray/_raylet.pyx

```diff
@@ -1722,6 +1732,25 @@ cdef class CoreWorker:
             self.job_config.ParseFromString(c_job_config.SerializeAsString())
         return self.job_config
 
+    def prepare_runtime_env(self, runtime_env_dict: dict) -> str:
+        """Update parent's runtime env with new env via a simple dict update.
+
+        If the resulting runtime env is empty, fall back to the runtime env
+        set in the JobConfig.  Returns the JSON-serialized runtime env.
+        """
+
+        result_dict = self.get_current_runtime_env_dict()
+        result_dict.update(runtime_env_dict)
+
+        # TODO(architkulkarni) remove once workers are cached by runtime env.
+        if all(val is None for val in result_dict.values()):
+            result_dict = {}
+
+        if result_dict == {}:
+            return self.get_job_config().serialized_runtime_env
+        else:
+            return json.dumps(result_dict, sort_keys=True)
```

Comment:
Ideally we'd construct another `RuntimeEnvDict` here so all the validation and serialization is done in one place